automation/test.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+#
+# This file is part of the KubeVirt project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2017 Red Hat, Inc.
+#
+
+# CI considerations: $TARGET is used by the jenkins build, to distinguish what to test
+# Currently considered $TARGET values:
+#     kubernetes-release: Runs all functional tests on a release kubernetes setup
+#     openshift-release: Runs all functional tests on a release openshift setup
+
+set -ex
+
+export WORKSPACE="${WORKSPACE:-$PWD}"
+readonly ARTIFACTS_PATH="exported-artifacts"
+
+mkdir -p "$ARTIFACTS_PATH"
+
+if [[ $TARGET =~ openshift-.* ]]; then
+  export KUBEVIRT_PROVIDER="os-3.10.0"
+elif [[ $TARGET =~ .*-1.10.4-.* ]]; then
+  export KUBEVIRT_PROVIDER="k8s-1.10.4"
+fi
+
+export KUBEVIRT_NUM_NODES=2
+
+kubectl() { cluster/kubectl.sh "$@"; }
+
+export CDI_NAMESPACE="${CDI_NAMESPACE:-kube-system}"
+
+make cluster-down
+make cluster-up
+
+# Wait for nodes to become ready
+set +e
+kubectl_rc=0
+while [ $kubectl_rc -ne 0 ] || [ -n "$(kubectl get nodes --no-headers | grep NotReady)" ]; do
```

Comment:
How do we handle timeouts here?

Reply:
We don't, I guess we could add a max retries, like 30, that way after 6 minutes we will abort.